daemon/src/model.rs

```diff
@@ -635,32 +638,131 @@ impl Default for FundingRate {
     }
 }
 
-/// Fee paid on every contract renewal (a fraction of it if rolling over)
-#[derive(Debug, Clone)]
+/// Fee paid by the taker to the maker for opening a cfd.
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Serialize, Deserialize)]
+pub struct OpeningFee {
+    #[serde(with = "::bdk::bitcoin::util::amount::serde::as_btc")]
+    amount: Amount,
+}
+
+impl OpeningFee {
+    pub fn new(amount: Amount) -> Self {
+        Self { amount }
+    }
+
+    /// Value in satoshis for the taker
+    pub fn to_u64(&self) -> u64 {
+        self.amount.as_sat()
+    }
+
+    pub fn amount_for(&self, role: Role) -> SignedAmount {
+        let amount_sat: i64 = self.amount.as_sat().try_into().expect("to convert");
+        let amount_sat = match role {
+            Role::Maker => -amount_sat,
+            Role::Taker => amount_sat,
+        };
+        SignedAmount::from_sat(amount_sat)
+    }
+}
+
+impl fmt::Display for OpeningFee {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.amount.fmt(f)
+    }
+}
+
+impl str::FromStr for OpeningFee {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let amount = Amount::from_str(s)?;
+        Ok(OpeningFee::new(amount))
+    }
+}
+
+impl Default for OpeningFee {
+    fn default() -> Self {
+        Self::new(Amount::ZERO)
+    }
+}
+
+impl_sqlx_type_display_from_str!(OpeningFee);
+
+/// Fee paid by the "Long" position
+#[derive(Clone, Copy)]
 pub struct FundingFee(u64);
 
 impl FundingFee {
     /// Value in satoshis
-    pub fn as_u64(&self) -> u64 {
+    pub fn to_u64(self) -> u64 {
         self.0
     }
 }
 
-impl Default for FundingFee {
+impl From<RolloverFee> for FundingFee {
+    fn from(funding_fee: RolloverFee) -> Self {
+        Self(funding_fee.to_u64())
+    }
+}
+
+impl From<OpeningFee> for FundingFee {
+    fn from(opening_fee: OpeningFee) -> Self {
+        // We calculate the fees for taker
+        Self(opening_fee.to_u64())
+    }
+}
+
+/// Fee paid on every contract renewal (a fraction of it if rolling over)
```

Comment:
I think this is incorrect. The `RolloverFee` is the accumulated previous funding fee (including opening fee as long as we do the bookkeeping as part of the payout calc)  + the current rollover fee.

If this is the only fee we pass in when rolling over the representation as fee + rate does not make sense.

Reply:
The different open ticket tracks these changes - this one is laying the groundwork for storing the fees.

Reply:
> Great start, I think the `RolloverFee` will need to be changed, but overall it looks good.

Does it need to be changed in this PR or can be done as a follow-up?